### DISCUSSION TITLE
Command 'git restore' not exposed

### DISCUSSION THREAD

**Discussion by lightningRalf** [Vinzent03/obsidian-git#739] [Q&A]
Hello together.
today paid off greatly using this awesome plugin! So thanks for that.
But I also had to venture outside of obsidian to achieve that.

Is there a reason why 'git restore' / or the older 'git checkout --' commands are not exposed?

Thanks in advance! 
Would greatly be appreshiated if they can be added. / Or if interested, I could try put in a PR with the help of gpt4o.

**Comment by Vinzent03** [comment]
You are right, seems like there is no dedicated command, but you should be able to accomplish this with the "Discard" button in the source control view pane.

**Reply by lightningRalf** [reply]
The source control view pane doesn't work as expected. Because it wants to commit everything every time. I did just stage one file. But it aborted, because of too many files to commit. (although I would have expected to only commit one)

In regards to the discard button:
![image](https://github.com/denolehov/obsidian-git/assets/126403501/89cb7560-cca6-4b34-9330-1bab91cbea92)
There is none for me.

---
I am using this plugin to just do git commands locally for the files within my vault. So no backup or other stuff is activated. (maybe this is a reason?)

**Reply by Vinzent03** [reply]
Are you on mobile or desktop? Please share a screenshot of the commit error. The discard button should be there for every file item in the list of changed files, not for all at the top.

**Reply by lightningRalf** [reply]
I am on desktop.
these are the two panes I found:
![image](https://github.com/denolehov/obsidian-git/assets/126403501/13046c4f-bb8d-429a-8f51-08a6f3a0addc)

I am a bit irritated of the last commit, because I did a bunch of commits on a different files via, `stage current file` and `git commit staged with specific message`.

When I do this now it shows the following messages:
![image](https://github.com/denolehov/obsidian-git/assets/126403501/2a34a1de-dba1-4bd9-91a1-86f1db75073b)


But when looking at the history (after hitting refresh) I get the following
![image](https://github.com/denolehov/obsidian-git/assets/126403501/55c2eb92-5073-4a41-a3ed-9ee0d11a5fd4)

**Reply by Vinzent03** [reply]
Seems like you have more than 500 changed files, which causes the files not to be displayed. You may use the terminal or another git application to handle those many changes. Maybe you just need to edit your `.gitignore`
